### PR TITLE
Explain Dockstore Tools and Workflows better

### PR DESCRIPTION
**Pull request by NatalieEO**
dockstore/dockstore#4542
https://docs.dockstore.org/en/feature-4542-describetoolsbetter/getting-started/intro-to-dockstore-tools-and-workflows.html

### CODE REVIEW

**Review by garyluu — APPROVED** [review]

File: docs/getting-started/intro-to-dockstore-tools-and-workflows.rst [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+written tool descriptor files (in CWL/WDL) that use it. At a basic level, the Docker image describes the tool environment and the descriptor files describe how the tool is run.
+If you are unfamiliar with Docker or how to write descriptor files, check out the following tutorials:
+
+- :doc:`Docker <getting-started-with-docker>`
+- :doc:`CWL <getting-started-with-cwl>`
+- :doc:`WDL <getting-started-with-wdl>`
+
+Dockstore:
+
+- has varying levels of support for images registered on Quay.io, DockerHub, GitLab, Amazon ECR, and Seven Bridges
```
Comment (coverbeck) [inline docs/getting-started/intro-to-dockstore-tools-and-workflows.rst:49]: and GCR I think

File: docs/getting-started/intro-to-dockstore-tools-and-workflows.rst [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+- :doc:`Hosted Tools <hosted-tools-and-workflows>`
+
+.. note::
+  Dockstore tool versions are based on the image's tags, not the tags/branches from the git repository where the descriptor files are hosted.
+
+
+Workflows
+---------
+
+Dockstore workflow registration is meant for users who have created or have access to descriptor files (in CWL, WDL, Galaxy, Nextflow). As mentioned above in the Tools section,
```
Comment (coverbeck) [inline docs/getting-started/intro-to-dockstore-tools-and-workflows.rst:69]: `or have access to descriptor files` -> the descriptor files in git technically (a local copy won't help you). But you cover it later and you'd have to do a bit of rewording, so you can leave it.

File: docs/getting-started/intro-to-dockstore-tools-and-workflows.rst [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+CWL and Galaxy classify tools and workflows differently, so only descriptor files written that follow language's respective specification for a workflow will be valid on Dockstore.
```
Comment (coverbeck) [inline docs/getting-started/intro-to-dockstore-tools-and-workflows.rst:70]: "...that follow language's..." -> "that follow the language's..."

**Review by coverbeck — APPROVED** [review]
This is really good. It helped me understand the difference better, as my behavior has been to generally ignore tools.

I think we should actively discourage users from registering WDL tools because they can't export them to Terra, but I don't think it belongs here. I vaguely remember discussing this at some storytime, and I forget if we were going to enforce that or not in code.

File: docs/getting-started/intro-to-dockstore-tools-and-workflows.rst (hunk 3 above)
Comment (aofarrel) [reply docs/getting-started/intro-to-dockstore-tools-and-workflows.rst:70]: I'd word it as 
> so only descriptor files written in a manner that follows a language's respective specification for a workflow will be valid on Dockstore.

**Review by aofarrel — COMMENTED** [review]
This is excellent and covers a lot of the most commonly asked questions about the tool-vs-workflow situation. I feel that some additions would provide further clarification -- I think we need to make it clear that a WDL workflow CAN have Docker image, especially as Terra requires this.

File: docs/getting-started/intro-to-dockstore-tools-and-workflows.rst [hunk 4]
```diff
@@ -0,0 +1,88 @@
+Introduction to Dockstore Tools and Workflows
+=============================================
+
+
+Comparison of Tools and Workflows
+---------------------------------
+
+When Dockstore was created, CWL was the first descriptor language we supported. CWL has a very clear distinction between a tool and a
+workflow. And while other descriptor languages, like WDL and Nextflow, do not have separate concepts for tools and workflows,
+Dockstore has chosen to treat the two differently. However, our definition for each does not completely align with the language's specification.
```
[the remark this answers is not in the document]
Comment (NatalieEO) [reply docs/getting-started/intro-to-dockstore-tools-and-workflows.rst]: Made it more general

**Review by kathy-t — APPROVED** [review]

**Review by aofarrel — APPROVED** [review]

**Comment by denis-yuen** [comment]
Late to the party, but thanks for doing this. I think it eliminates something that has been stuck on my plate

**Pull request merged**